### PR TITLE
build: add release workflow

### PR DESCRIPTION
**Pull request by dmccartney**
This adds a workflow to publish builds to pub.dev.

### CODE REVIEW

**Review by elisealix22 — APPROVED** [review]
🥲

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  dart:
+    name: Release (dart)
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dart-lang/setup-dart@v1.3
+      - run: dart pub get
+      - run: dart test
+      - run: dart pub publish --dry-run
```
Comment (elisealix22) [inline .github/workflows/release.yml:15]: @dmccartney should we drop the --dry-run? Ready to test this thing out!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,3 @@
-## 0.0.1
+## 0.0.1-development
 
-* TODO: Describe initial release.
+- initial development pre-release
```
Comment (elisealix22) [inline CHANGELOG.md:3]: Nit: We landed on the name `Developer Preview` might be nice to sync that here!

File: .github/workflows/release.yml (hunk 1 above)
Comment (dmccartney) [reply .github/workflows/release.yml:15]: I was thinking to separate (for now) getting the pub.dev credential secret configured etc.
(We'll want it for [the proto dart package](https://github.com/xmtp/proto/blob/main/.github/workflows/release.yml#L19-L21) too )

**Pull request merged**